### PR TITLE
Remember columns width

### LINKED ISSUE #444: The resized width of columns will be reset after sorting
## Bug Report

The resized width of columns will be reset after sorting, we should remember their width.

### PR DESCRIPTION
**Pull request by baurine**
resolve #444 

What did:

- Use ShimmeredDetailsList to represent loading status for not re-mount the DatailsList when reloading data
- Keep changed columns width when reloading data in the current page
- Show the existed old data when reloading

Failed to do:

- Persist the width of the columns.

  According to the issue https://github.com/microsoft/fluentui/issues/9287 , this feature isn't supported by DatailsList.

Do you guys have any idea?

### CODE REVIEW

**Comment by breeswish** [comment]
Actually if you only want to prevent the re-mount, maybe consider `display: none`?

**Comment by baurine** [comment]
> Actually if you only want to prevent the re-mount, maybe consider `display: none`?

Yep, that's a way! but I also prefer to show loading status inside the table a bit.

File: ui/lib/apps/Statement/pages/List/index.tsx [hunk 1]
```diff
@@ -52,6 +54,22 @@ export default function StatementsOverview() {
...
+    newWidth?: number,
+    _columnIndex?: number
+  ) {
+    setColumnsWidth({
+      ...columnsWidth,
+      [column!.key]: newWidth,
+    })
+  }
+  const { run } = useDebounceFn(handleColumnResize, 500)
+
```
Comment (baurine) [inline ui/lib/apps/Statement/pages/List/index.tsx:72]: These don't work code will be reverted in the next PR.

File: ui/lib/components/CardTableV2/index.tsx [hunk 2]
```diff
@@ -78,12 +79,30 @@ function CardTableV2(props: ICardTableV2Props) {
...
+                width: `${columnsWidth[c.key]}px`,
+              }, // doesn't work
+              currentWidth: columnsWidth[c.key], // doesn't work
+              calculatedWidth: columnsWidth[c.key], // doesn't work
+            }
+          : c
+      )
+    }
+    console.log(newColumns)
+    return newColumns
```
Comment (baurine) [inline ui/lib/components/CardTableV2/index.tsx:104]: These don't work code will be reverted in the next PR.

**Pull request merged**